### ISSUE TITLE
Get worng defaut vendor when you create " a price request " and " purchase order " from projet

### ISSUE THREAD

**Issue by trekmorris** [Dolibarr/dolibarr#28248]
### Bug

When you want to create a " a price request " and " purchase order " from projet , then the default vendor will be thirtyparty himself .
![image](https://github.com/Dolibarr/dolibarr/assets/78349579/91f4fe68-fda4-4796-a249-b0f0143bedfb)
![image](https://github.com/Dolibarr/dolibarr/assets/78349579/c3b9909a-6ca6-4b6f-9c16-819df8c8aeb7)
 

### Environment Version

v18.0.5

### Environment OS

Deian 11

### Environment Web server

Apache/2.4.38 (Debian)

### Environment PHP

 8.1.27

### Environment Database

MariaDB 10.3.39-MariaDB-0+deb10u2

### Environment URL(s)

/projet/card.php?id=885

### Expected and actual behavior

Should could select the vendor who you want .

### Steps to reproduce the behavior

Create a new project for a customer then create a purchase order by click " create purchase order " in main Project tab .

**Comment by JonBendtsen** [comment]
yes, seems like even in 19.0.0 no matter which of the categories (ecvept for expense report or donation) are done with the vendor of a project.

![wrong_default_vendor](https://github.com/Dolibarr/dolibarr/assets/2551481/f55b9d3c-a5e7-4794-a6cc-012d077d49cc)

**Comment by JonBendtsen** [comment]
I wonder if this depends on whether or not the thirdparty assigned to the project was also configured as a vendor

**Comment by JonBendtsen** [comment]
> I wonder if this depends on whether or not the thirdparty assigned to the project was also configured as a vendor

we should check this

**Comment by pixodeo** [comment]
On 21.0 the "Create supplier order" button on a project (`htdocs/projet/card.php:1756`) only passes `socid=...` when the project's thirdparty is exclusively a supplier (`$is_supplier_only` = is a fournisseur AND not a customer/prospect). When the thirdparty is a customer or prospect, the URL is just `?action=create&projectid=...`, so `htdocs/fourn/commande/card.php` opens with no `socid` and the Supplier dropdown is empty by default (it is filtered on `s.fournisseur=1 AND s.status=1` at line 2004).

Could you confirm whether the thirdparty linked to the failing project is also flagged as a supplier in its third-party card? @JonBendtsen raised exactly that hypothesis in April 2026. If yes, the current code passes the customer as the supplier, which is the bug behavior you describe. If not, the prefill is coming from somewhere else and a fresh trace from a current version would help (URL of the page, browser network tab, dolibarr.log of the click).

If it confirms the dual-role hypothesis, the minimum-diff fix is to drop the `socid` pass-through for the supplier_proposal / supplier_order / supplier_invoice buttons whenever the thirdparty is both a customer/prospect and a supplier, since in that case there is no unambiguous default and the user should pick a vendor explicitly. Happy to open a PR with that change once the dual-role case is confirmed as the trigger.

**Comment by pixodeo** [comment]
Could you confirm whether this still reproduces on 23.0.x? Reading the current code on 23.0:

- htdocs/projet/card.php:1753 builds the 'AddSupplierOrder' link with /fourn/commande/card.php?action=create&projectid=N - it intentionally drops the socid unless the third party is supplier-only.
- htdocs/fourn/commande/card.php:1755-1764 renders the Supplier field with select_company filtered on s.fournisseur=1; with empty $socid the dropdown should be empty (not preselected).
- htdocs/projet/element.php:591 (price request link) has the same pattern.

If the screenshot still applies on 23.0, please share the URL of the create page after you click the button (right after the redirect) - it will tell us whether socid is being injected somewhere I missed.
